standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -604,15 +630,35 @@ public OpenTxnsResponse openTxns(OpenTxnRequest rqst) throws MetaException {
   @Override
   @RetrySemantics.Idempotent
   public void abortTxn(AbortTxnRequest rqst) throws NoSuchTxnException, MetaException, TxnAbortedException {
-    long txnid = rqst.getTxnid();
+    long txnid;
+    long sourceTxnId = -1;
+    ResultSet rs;
     try {
       Connection dbConn = null;
       Statement stmt = null;
       try {
         lockInternal();
         dbConn = getDbConn(Connection.TRANSACTION_READ_COMMITTED);
+        stmt = dbConn.createStatement();
+
+        if (rqst.isSetReplPolicy()) {
+          sourceTxnId = rqst.getTxnid();
+          String query = "select TM_TARGET_TXN_ID from REPL_TXN_MAP where TM_SRC_TXN_ID = " + sourceTxnId +
+              " and TM_REPL_POLICY = " + quoteString(rqst.getReplPolicy());
+          String s = sqlGenerator.addForUpdateClause(query);
+          LOG.debug("Going to execute query <" + s + ">");
+          rs = stmt.executeQuery(s);
+          if (!rs.next()) {
+            LOG.debug("Target txn for Transaction not present <" +
```

Comment:
Add info log as it happens for idempotent case. 
Also, shall re-frame the statement as "Target txn is missing for the input source txn (ReplPolicy: <dbName.tblName>, srcTxnId: <sourceTxnId>)

Reply:
fixed